tests/backup/backup_test.go

```diff
@@ -143,6 +143,110 @@ var _ = Describe("{UserGroupManagement}", func() {
 	})
 })
 
+// This is to create multiple users and groups
+var _ = Describe("{CreateMultipleUserAndGroup}", func() {
+	numberOfUsers := 20
+	numberOfGroups := 10
+	users := make([]string, 0)
+	groups := make([]string, 0)
+	groupNotCreated := make([]string, 0)
+	userNotCreated := make([]string, 0)
+
+	JustBeforeEach(func() {
+		wantAllAfterSuiteActions = false
+		StartTorpedoTest("CreateMultipleUserAndGroup", "Creation of multiple users and groups", nil, 0)
+	})
+	It("Create multiple users and Group", func() {
+
+		Step("Create Groups", func() {
+			log.InfoD("Creating %d groups", numberOfGroups)
+			var wg sync.WaitGroup
+			for i := 1; i <= numberOfGroups; i++ {
+				groupName := fmt.Sprintf("testGroup%v", i)
+				wg.Add(1)
+				go func(groupName string) {
+					err := backup.AddGroup(groupName)
+					log.FailOnError(err, "Failed to create group - %v", groupName)
+					groups = append(groups, groupName)
+					wg.Done()
+				}(groupName)
+			}
+			wg.Wait()
+		})
+
+		Step("Create Users", func() {
+			log.InfoD("Creating %d users", numberOfUsers)
+			var wg sync.WaitGroup
+			for i := 1; i <= numberOfUsers; i++ {
+				userName := fmt.Sprintf("testuser%v", i)
+				firstName := fmt.Sprintf("FirstName%v", i)
+				lastName := fmt.Sprintf("LastName%v", i)
+				email := fmt.Sprintf("testuser%v@cnbu.com", i)
+				wg.Add(1)
+				go func(userName, firstName, lastName, email string) {
+					err := backup.AddUser(userName, firstName, lastName, email, "Password1")
+					log.FailOnError(err, "Failed to create user - %s", userName)
+					users = append(users, userName)
+					wg.Done()
+				}(userName, firstName, lastName, email)
+			}
+			wg.Wait()
+		})
+
+		//This loops through the groups that we have created againts the groups fetched and checks if group is present
+
+		Step("Validate Group", func() {
+			createdgroups, err := backup.GetAllGroups()
+			log.Info("created group ", createdgroups)
+			log.FailOnError(err, "Failed to get group")
+			for _, group := range groups {
+				found := false
+				for _, createdgroup := range createdgroups {
```

Comment:
Better create a map of groups and traverse it rather than having loop running multiple times

Reply:
made changes where it creates response map and  iterates through the group names slice and checks if the group name is present in the structs map using map[key] to get the value and key to check whether the it is present or not. If it's not found, it prints the group name not found in struct slice.

now the method is O(n) time complexity in total, where n is the length of group names slice